runtime/queries/nice.go

```diff
@@ -0,0 +1,52 @@
+package queries
+
+import (
+	"math"
+)
+
+var (
+	e10 = math.Sqrt(50)
+	e5  = math.Sqrt(10)
+	e2  = math.Sqrt(2)
+)
+
+func calculateE(stepError float64) float64 {
+	if stepError >= e10 {
+		return 10
+	} else if stepError >= e5 {
+		return 5
+	} else if stepError >= e2 {
+		return 2
+	}
+	return 1
+}
+
+func tickIncrement(start, stop, count float64) float64 {
+	step := (stop - start) / math.Max(0, count)
+	power := math.Floor(math.Log(step) / math.Log(10))
+	stepError := step / math.Pow(10, power)
+
+	e := calculateE(stepError)
+	if power >= 0 {
+		return e * math.Pow(10, power)
+	}
+
+	return -math.Pow(10, -power) / e
+}
+
+func NiceAndStep(start, stop, count float64) []float64 {
```

Comment:
1. Please add a comment to the function explaining what it does
2. I don't feel certain this loop is always going to terminate in all cases. Especially since it relies on comparing float64s. Is there any way we can make it bounded? (Even if just an upper bound.)
3. Make it a private function since it's not used outside of the package
4. Either move it and its tests to the `column_numeric_histogram` file or rename the file to something like `column_numeric_histogram_xxx` to clarify relationship (the `queries` package is quite big)

Reply:
3. It's used in `queries` package too.